### PR TITLE
[Edge Stack 2.3.0] Fixes for Edge Stack 2.3.0

### PR DESCRIPTION
**Pull request by LukeShu**
## Description
This fixes some things that affect the Edge Stack RC, but don't warrant cutting a new Emissary RC.

## Testing
This is tested as part of testing apro.

## Checklist

 - [x] I made sure to update `CHANGELOG.md`. - no applicable changes (in this repo)

   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations

 - [x] This is unlikely to impact how Ambassador performs at scale.

   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability

 - [ ] My change is adequately tested. - kinda

   Remember when considering testing:
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points

 - [x] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently. - no tricks

 - [ ] The changes in this PR have been reviewed for security concerns and adherence to security best practices. - what?

### CODE REVIEW

File: cmd/entrypoint/testutil_fake_syntheticauth_test.go [hunk 1]
```diff
@@ -574,3 +574,158 @@ spec:
...
+	assert.Equal(t, 1, len(snap.Kubernetes.AuthServices))
+
+	for _, authService := range snap.Kubernetes.AuthServices {
+		assert.Equal(t, "dummy-service", authService.Spec.AuthService)
+	}
+
+	// Check for an ext_authz cluster name matching the provided AuthService (Http_Filters are harder to check since they always have the same name)
+	// the namespace for this extauthz cluster should be foo (since that is the namespace of the valid AuthService above)
+	isAuthCluster := func(c *v3cluster.Cluster) bool {
+		return strings.Contains(c.Name, "cluster_extauth_127_0_0_1_8500_foo")
```
Comment (AliceProxy) [inline cmd/entrypoint/testutil_fake_syntheticauth_test.go:675]: I was actually missing a call to `f.Flush()` after applying the above yaml for this test, and `cluster_extauth_127_0_0_1_8500_foo` should have been `cluster_extauth_dummy_service_foo`.

You can just switch [da11c1a](https://github.com/emissary-ingress/emissary/pull/4264/commits/da11c1ad20310086e28353964f02bda281550ae7) to [552b9aa](https://github.com/emissary-ingress/emissary/commit/552b9aa747a9147dd566c6d03a85f5aa12b2fef9) and that will resolve the test failure.

Comment (LanceEa) [reply cmd/entrypoint/testutil_fake_syntheticauth_test.go:675]: Is this still relevant? Is it for @LukeShu

**Review by LanceEa — APPROVED** [review]
All looks good, You have one outstanding comment that I think is for Luke or mental reminder?

**Comment by LukeShu** [comment]
I'm leaving this as a draft to prevent accidental merging because it isn't worth merging before we either (1) cut 2.3.0 GA or (2) decide we need to do another RC.

**Comment by LukeShu** [comment]
Marking as ready for review now that 2.3.0 GA has been released.

**Review by LanceEa — APPROVED** [review]
lgtm

**Pull request merged**